src/pipelinedefinition.cpp

```diff
@@ -477,6 +477,21 @@ class NodeValidator {
         return StatusCode::OK;
     }
 
+    bool isConnectionShapeCorrect(const shape_t& tensorInputShape, const shape_t& tensorOutputShape) {
```

Comment:
I would change the name. I can see that this method provides comparison logic for tensor shapes. 
For easier reuse in other places maybe we could name it: areShapesEqual() or something like that to make it more generic?